src/ProxyHandler.cc

```diff
@@ -3,16 +3,159 @@
 const std::string HOST_TOKEN = "host";
 const std::string PORT_TOKEN = "port";
 
+using boost::asio::ip::tcp;
+
 RequestHandler::Status ProxyHandler::HandleRequest(const Request& request,
                      Response* response) {
-    response->SetStatus(Response::ok);
-    response->SetVersion("HTTP/1.0");
-    response->AddHeader("Content-Type", "text/plain");
-    response->SetBody(request.raw_request());
+
+    // Get endpoints corresponding to the server name.
+    boost::asio::io_service io_service;
```

Comment:
I would suggest put sending and receiving http packet logic into another class. 
Later when you need to handle redirect, you will need to do the same thing, sending and receiving http packet, lots of time. Wrapping them into a class will make the codes look more clean